### PR TITLE
Adding delayed spotlight to feedback button on current product page

### LINKED ISSUE #35301: CES: Add prompt with spotlight after 7 minutes to create product
Display a spotlight near the Feedback button ([added here](https://github.com/woocommerce/woocommerce/issues/35300)) when the user takes more than 7 mins to create a product on the classic UI.

image

Update the copy to read:
> You have been working on this product for a few minutes now. Is there something you’re struggling with? Share your feedback.

See additional context on [this](https://github.com/woocommerce/woocommerce/issues/35127#issuecomment-1282884238) (and following) comments, and note the feedback button is added on [this issue](https://github.com/woocommerce/woocommerce/issues/35806).

Acceptance criteria:
- Spotlight appears after user takes 7 minutes to create a product (independent of user interaction)
- It is only displayed once

### PR DESCRIPTION
**Pull request by joelclimbsthings**
-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This will create a spotlight component to highlight the feedback button on the current product UI, only after 7 minutes. 

- User interaction doesn't matter--if they're still on the page it will be displayed. 
- It will only be displayed once.

### Screenshots

image

Closes #35301 .

### How to test the changes in this Pull Request:

1. Checkout branch on fresh store.
2. The timeout is set to 7 minutes. You can either wait that amount of time, or [alter this line](https://github.com/woocommerce/woocommerce/blob/d9a577859cedf651efa67ed233dac0887a2f38b9/plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx#L11) to a more reasonable timeout for testing.
3. Navigate to current product UI (Products -> Add new)
4. Notice Feedback button on Activity bar at top right. After the timeout elapses you should see the spotlight popover appear as pictured above.
5. Refresh, and wait. It should not appear again.
6. Delete the `woocommerce_ces_product_feedback_shown` option on your store, and refresh. It should appear again as it did originally.
7. Click the "Feedback" button and either close the modal or fill it out and submit. The popover should close at the same time.
8. Delete the option and refresh. Immediately hit the "Feedback" button (before popover displays), and close. The popover should not display.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+const FEEDBACK_TOUR_OPTION = 'woocommerce_ces_product_feedback_shown';
+const FEEDBACK_TIMEOUT_MS = 7 * 60 * 1000;
+
+const useShowProductFeedbackTour = (): undefined | boolean => {
+	const { tourOptionValue } = useSelect( ( select ) => {
+		const { getOption } = select( OPTIONS_STORE_NAME );
+
+		return {
+			tourOptionValue: getOption( FEEDBACK_TOUR_OPTION ) as
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx]: This isn't an issue we can handle here and I think you made the only logical choice, but just noting that we're undoing all the work to remove `getOption` calls lately.

@woocommerce/mothra I think we should seriously consider introducing some kind of dismissal API before we continue to introduce a lot more debt around this options API.

File: plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+};
+
+type ProductFeedbackTourProps = {
+	currentTab: string;
+};
+
+export const ProductFeedbackTour: React.FC< ProductFeedbackTourProps > = ( {
+	currentTab,
+} ) => {
+	const tourOptionValue = useShowProductFeedbackTour();
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx]: Nitpicky, but it took me a second to figure out what `tourOptionValue`.  Should this name hint at what this value means? Maybe `hasShownTour` or something similar?

**Review by joshuatf — COMMENTED** [review]
This is testing well for me on the new product pages; nice work, Joel!

I ran into an issue on an existing product page where the modal is still shown but in the bottom left:
Screen Shot 2022-12-07 at 1 07 30 PM

File: plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx (hunk 2 above)
Comment (joelclimbsthings) [reply plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx]: Hey, we're all professional nitpickers here! Done in https://github.com/woocommerce/woocommerce/pull/35865/commits/6e231063d25268a297e844328532a0860197f0de

File: plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx (hunk 1 above)
Comment (joelclimbsthings) [reply plugins/woocommerce-admin/client/guided-tours/add-product-feedback-tour.tsx]: Agreed 👍🏻

**Comment by joelclimbsthings** [comment]
> I ran into an issue on an existing product page where the modal is still shown but in the bottom left:

Whoops! The intention here is actually to _only_ show this on the classic product page, but I lost that little bit of logic on a refactor; thanks @joshuatf ! I've fixed this in https://github.com/woocommerce/woocommerce/pull/35865/commits/58d7ea0bdbd6976074490ace59b4cf897648b3c9

The weird part is that it looks like the spotlight was displayed improperly on your local for the classic product page. My guess would be that you set the timeout _too_ low, so that the feedback button hadn't been added to the DOM before it triggered. I'd try setting it to a second or two at least.

**Review by joshuatf — DISMISSED** [review]
Thanks for the changes, Joel!

>The weird part is that it looks like the spotlight was displayed improperly on your local for the classic product page. My guess would be that you set the timeout too low, so that the feedback button hadn't been added to the DOM before it triggered. I'd try setting it to a second or two at least.

It was set at 7s, but I think the issue from the screenshot was that it was on an edit product page (not add) which was resolved by https://github.com/woocommerce/woocommerce/commit/58d7ea0bdbd6976074490ace59b4cf897648b3c9.

All is testing well for me now!

**Review by joshuatf — APPROVED** [review]
Still tests well.  Thanks for the rebase! :shipit:

**Pull request merged**